src/IntervalComparator.php

```diff
@@ -140,26 +140,36 @@ public function cover(ComparableIntervalInterface $interval)
     {
         $type = IntervalType::TYPE_CLOSED;
 
-        if ($this->interval->startPoint()->lt($interval->startPoint())) {
+        if ($this->interval->startPoint()->eq($interval->startPoint())) {
+            $start = $this->interval->startPoint();
+            if ($this->interval->type()->startExcluded() && $interval->type()->startExcluded()) {
+                $type |= IntervalType::TYPE_START_EXCLUDED;
+            }
+        } elseif ($this->interval->startPoint()->lt($interval->startPoint())) {
             $start = $this->interval->startPoint();
             if ($this->interval->type()->startExcluded()) {
                 $type |= IntervalType::TYPE_START_EXCLUDED;
             }
         } else {
             $start = $interval->startPoint();
-            if ($interval->type()->startExcluded()) {
+            if ($this->interval->startPoint()->neq($start) && $interval->type()->startExcluded()) {
```

Comment:
`$this->interval->startPoint()->neq($start)` always return `false` here.

Reply:
This is big truth!

I see, ruled that out above.